File: 0129. Sum Root to Leaf Numbers.py
from typing import Optional

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        return self.dfs(root, '')

    def dfs(self, node: Optional[TreeNode], seen: str) -> int:
        if node.left is None and node.right is None:
            return int(seen + str(node.val))
        result = 0
        if node.left:
            result += self.dfs(node.left, seen + str(node.val))
        if node.right:
            result += self.dfs(node.right, seen + str(node.val))
        return result
        

File: 0129. Sum Root to Leaf Numbers.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun sumNumbers(root: TreeNode?): Int {
        return dfs(0, root)
    }
    fun dfs(num: Int, root: TreeNode?): Int {
        if (root == null) return 0
        var num = (num * 10) + root.`val`
        if (root.left == null && root.right == null) return num
        return dfs(num, root.left) + dfs(num, root.right)
    }
}